### PR TITLE
Selinux mode toggle

### PR DESCRIPTION
**Pull request by shpsec-david**
I'm not as confident about this as I was about the last set of changes.

This would help address https://github.com/saltstack/salt/issues/8640 as well as (I believe) properly address what people intend to do when changing SELinux modes.

The core issue is the current reliance on the 'enforce' file, which is not existent if SELinux is Disabled.  Even if the system is capable and all prerequisites are covered, the file only exists when SELinux is in Permissive or Enforcing modes.

The changes here would keep the toggle between Permissive/Enforcing, but would also allow changing to/from Disabled mode.  Doing so does require a reboot, and I'm leaving that up to the end user.

Also by making additional changes to the /etc/selinux/config file, the changes would persist across reboots, modifying the contents of /sysfs/selinux/enforce does not do so.

### CODE REVIEW

File: salt/modules/selinux.py [hunk 1]
```diff
@@ -109,13 +117,31 @@ def setenforce(mode):
             mode = '0'
     else:
         return 'Invalid mode {0}'.format(mode)
-    enforce = os.path.join(selinux_fs_path(), 'enforce')
+
+    if getenforce() != 'Disabled': # enforce file does not exist if currently disabled.  Only for toggling enforcing/permissive
```
Comment (cachedout) [inline salt/modules/selinux.py]: Please add one more space here before the `#`.

**Comment by thatch45** [comment]
Very nicely done @shpsec-david, this covers the bases well, thanks!

**Comment by shpsec-david** [comment]
Excellent.  Glad to see you folks are happy with the changes.

**Comment by shpsec-david** [comment]
Checking in on this.  I thought it was in early enough to make the 2016.3 cutoff, but it doesn't appear to have.  And my git-fu is not strong enough to trace the trail and see if it just missed the date, or if it actually got lost along the way.

**Pull request merged**